src/gui/qgsfilewidget.cpp

```diff
@@ -281,7 +281,7 @@ void QgsFileWidget::openFileDialog()
 
   // if we use a relative path option, we need to obtain the full path
   // first choice is the current file path, if one is entered
-  if ( !mFilePath.isEmpty() )
+  if ( !mFilePath.isEmpty() && QFile( mFilePath ).exists() )
```

Comment:
```suggestion
  if ( !mFilePath.isEmpty() && QFile::exists( mFilePath ) )
```

(the static QFile::exists() method is more efficient vs constructing a QFile object)